pkgs/servers/sql/cockroachdb/default.nix

```diff
@@ -1,70 +1,52 @@
-{ lib, stdenv, buildGoPackage, fetchurl
-, cmake, xz, which, autoconf
-, ncurses6, libedit, libunwind
-, installShellFiles
-, removeReferencesTo, go
+{ lib
+, stdenv
+, buildFHSEnv
 }:
 
 let
-  darwinDeps = [ libunwind libedit ];
-  linuxDeps  = [ ncurses6 ];
-
-  buildInputs = if stdenv.isDarwin then darwinDeps else linuxDeps;
-  nativeBuildInputs = [ installShellFiles cmake xz which autoconf ];
-
-in
-buildGoPackage rec {
-  pname = "cockroach";
+  name = "cockroach";
   version = "20.1.8";
 
-  goPackagePath = "github.com/cockroachdb/cockroach";
-
-  src = fetchurl {
-    url = "https://binaries.cockroachdb.com/cockroach-v${version}.src.tgz";
-    sha256 = "0mm3hfr778c7djza8gr1clwa8wca4d3ldh9hlg80avw4x664y5zi";
-  };
-
-  env.NIX_CFLAGS_COMPILE = toString (lib.optionals stdenv.cc.isGNU [ "-Wno-error=deprecated-copy" "-Wno-error=redundant-move" "-Wno-error=pessimizing-move" ]);
-
-  inherit nativeBuildInputs buildInputs;
+  # For several reasons building cockroach from source has become
+  # nearly imposible. See https://github.com/NixOS/nixpkgs/pull/152626
+  # Therefore we use the pre-build release binary and wrap it with buildFHSUserEnv to
+  # work on nix.
+  # You can generate the hashes with
+  # nix-prefetch-url --type sha256 --unpack ...
+  src = fetchTarball (lib.getAttr stdenv.system {
+    x86_64-linux = {
+      name = "cockroach-release-${version}";
+      url = "https://binaries.cockroachdb.com/cockroach-v${version}.linux-amd64.tgz";
+      sha256 = "1ra2l02psy449hay89nh87qwh3rspfcfb5i9r9gasd0qx8ybnbpc";
```

Comment:
SRI hashes, please.

Reply:
Now that I know of `store prefetch-file <url>`, with pleasure.